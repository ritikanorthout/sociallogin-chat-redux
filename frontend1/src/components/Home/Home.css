.detail-container {
  background-color: #f4f5f7;
  height: 100vh;
}

.containers {
  width: 500px;
  position: absolute;
  height: 400px;
  margin: 20px 10px;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: white;
  text-align: center;
  border-style: groove;
  border-color: #f4f5f7;
}
.containers img {
  margin-top: 40px;
  border-radius: 12px;
}


.containers2 {
  width: 500px;
  background-color: white;
  text-align: center;
  border-style: groove;
  border-color: #f4f5f7;
}

.text {
  font-size: 25px;
}
