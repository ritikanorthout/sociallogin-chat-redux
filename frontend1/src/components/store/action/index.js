import { AUTH_LOGIN, USER_REGISTER } from "../../constant/type";

// export const fngetLogin = (data) => {
//   return (dispatch) => {
//     dispatch({
//       type: AUTH_LOGIN,
//       payload: data,
//     });
//   };
// };

// export const fnuserRegister = (userData) => {
//   return (dispatch) => {
//     dispatch({
//       type: USER_REGISTER,
//       payload: userData,
//     });
//   };
// };

export const getLogin = (data) => {
  return (dispatch) => {
    dispatch({
      type: AUTH_LOGIN,
      payload: data,
    });
  };
};

export const userRegister = (userData) => {
  return (dispatch) => {
    dispatch({
      type: USER_REGISTER,
      payload: userData,
    });
  };
};
