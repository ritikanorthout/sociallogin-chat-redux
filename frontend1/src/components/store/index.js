import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import Reducer from "./reducer";

function saveToLocalStorage(state) {

  console.log('saveToLocalStorage ', state);

  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem("state", serializedState);
  } catch (error) {
    console.log(error);
  }
}

function loadToStore() {

  console.log('loadToStore');
  try {
    const serializedState = localStorage.getItem("state");
    console.log('serializedState',serializedState);
    if (serializedState === null) return undefined;
    return JSON.parse(serializedState);
  } catch (error) {
    console.log(error);
  }
}
const per = loadToStore();
const store = createStore(Reducer, per, applyMiddleware(thunk));

store.subscribe(() => saveToLocalStorage(store.getState()));
export default store;
