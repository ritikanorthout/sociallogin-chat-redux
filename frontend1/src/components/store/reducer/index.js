import { AUTH_LOGIN, USER_REGISTER, LOGOUT } from "../../constant/type";
const initialState = {
  isLoggedin: false,
  userInfo: { name: "", email: "", mobile: "", token: "", imgUrl: "" },
  userList: [],
};

const Reducer = (state = initialState, action) => {
  console.log("action.payload : ",action.payload);

  console.log("state : ",state);

  switch (action.type) {
    case AUTH_LOGIN:
      return {
        ...state,
        isLoggedin: true,
        userInfo: {
          name: action.payload.name,
          email: action.payload.email,
          imgUrl: action.payload.imageUrl,
        },
      };
    case USER_REGISTER:
      return {
        ...state,
        userList: [...state.userList, action.payload],
      };
    case LOGOUT:
      return {
        ...state,
        userInfo: { name: "", email: "", mobile: "", token: "", imgUrl: "" },
        isLoggedin: false,
      };
    default:
      return {
        ...state,
      };
  }

};

export default Reducer;
