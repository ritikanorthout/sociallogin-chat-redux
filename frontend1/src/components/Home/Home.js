import React from "react";
import "./Home.css";
import { Card, CardBody, CardTitle, CardSubtitle, CardText, Button } from "reactstrap";
import Header from "../Header/Header";
import Footer from "../Footer/Footer";
import Chat from "../Chat/Chat";

function Home(props) {
  console.log("props", props);

  return (
    <div className="detail-container">
      <Header />
      <div className="text">Welcome {props.data.userInfo.name}
      </div>
      <Card className="containers">
        {(props.data.userInfo.imgUrl) &&
        <CardBody>
          <img
            width="200px"
            height="200px"
            src={props.data.userInfo.imgUrl}
            alt="Profle Pic"
          />
        </CardBody>
        }

        <CardBody>
          <CardTitle className="text">
            Name : {props.data.userInfo.name}
          </CardTitle>
          {props.data.userInfo.email ? (
            <p className="text">Email : {props.data.userInfo.email}</p>
          ) : null}
        </CardBody>
  
      </Card>
      <div>
      <Card className="containers2">
        {/* <CardImg top width="100%" src="/assets/318x180.svg" alt="Card image cap" /> */}
        <CardBody>
          <Chat name={props.data.userInfo.name}/>
        </CardBody>
      </Card>
    </div>

      <Footer />
    </div>
  );
}

export default Home;
