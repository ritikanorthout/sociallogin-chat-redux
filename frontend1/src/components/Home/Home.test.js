import React from "react";
import ReactDOM from "react-dom";
import Home from "./Home";
import { Provider } from "react-redux";
import store from "../store";

it("Home renders without crashing", () => {
  const initialState = {
    data: {
      isLoggedin: false,
      userInfo: { name: "", email: "", mobile: "", token: "", imgUrl: "" },
      userList: [],
    },
  };
  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <Home {...initialState} />
    </Provider>,
    div
  );

  expect(div).toMatchSnapshot();
});
