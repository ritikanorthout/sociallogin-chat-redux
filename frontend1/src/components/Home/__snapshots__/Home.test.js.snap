// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
<div>
  <div
    class="detail-container"
  >
    <div>
      <div
        class="callout primary"
        id="Header"
      >
        <div
          class="row column"
        >
          <button
            class="logouButton"
          >
            logout
             
          </button>
        </div>
      </div>
    </div>
    <div
      class="text"
    >
      Welcome 
      
    </div>
    <div
      class="containers card"
    >
      <div
        class="card-body"
      >
        <img
          alt="Profle Pic"
          height="200px"
          src=""
          width="200px"
        />
      </div>
      <div
        class="card-body"
      >
        <div
          class="text card-title"
        >
          Name : 
          
        </div>
      </div>
    </div>
    <div>
      <div
        class="row"
        id="footer"
      >
        <div
          class="medium-12 columns"
        >
          <p />
        </div>
      </div>
    </div>
  </div>
</div>
`;
